presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1115,14 +1117,14 @@ public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTa
                 ImmutableList.of());
     }
 
-    private static TupleDomain<ColumnHandle> createPredicate(List<ColumnHandle> partitionColumns, List<HivePartition> partitions)
+    @VisibleForTesting
+    static TupleDomain<ColumnHandle> createPredicate(List<ColumnHandle> partitionColumns, List<HivePartition> partitions)
     {
         if (partitions.isEmpty()) {
             return TupleDomain.none();
         }
 
-        Map<ColumnHandle, Domain> domains = new HashMap<>(partitionColumns.size());
-
+        Multimap<ColumnHandle, NullableValue> partitionToValues = HashMultimap.create();
```

Comment:
I haven't tried writing it that way, but it may be cleaner the other way. I don't think performance will matter in practice. You'd probably need to have hundreds of thousands of partitions to see a meaningful difference, and that's not the common case.